ldap-api/src/main/java/ca/bc/gov/hlth/ldapapi/service/LdapService.java

```diff
@@ -83,6 +87,36 @@ private boolean checkUserLocked(Attributes attributes) {
         return unlocked;
     }
 
+    private boolean checkUserPasswordExpired(Attributes attributes) {
+        Attribute passwordChangeDate = attributes.get(LDAP_ATTR_PASSWORD_CHANGE_DATE);
+        Attribute passwordLifespan = attributes.get(LDAP_ATTR_PASSLIFESPAN);
+        boolean expired = false;
+
+        if (passwordChangeDate != null) {
+            try {
+                Long changedDaysAgo = Long.valueOf(passwordChangeDate.get().toString());
+                Long today = LocalDate.now().toEpochDay();
+                expired = (changedDaysAgo + checkPasswordLifespan(passwordLifespan)) < today;
```

Comment:
Just confirming, but the date we store on LDAP is also an "Epoch Day" so this is a valid comparison?

Reply:
Yep.